From cf1b8ce583d3dbbd8988d1c1b3c6d5c3ff4dc8d0 Mon Sep 17 00:00:00 2001
From: Adeel Arshad <adeel_arshad@mentor.com>
Date: Fri, 7 Oct 2016 15:35:53 +0500
Subject: [PATCH] set the CPPFLAGS for cross compile

Prefix the include directories with the sysroot directory,
so that it would become compatible for cross compilation.

Signed-off-by: Adeel Arshad <adeel_arshad@mentor.com>
---
 CodeXL/Components/Graphics/GPS_init.py | 20 ++++++++++----------
 Common/Src/SCons/CXL_init.py           | 22 +++++++++++-----------
 2 file changed, 21 insertions(+), 21 deletions(-)

diff --git a/CodeXL/Components/Graphics/GPS_init.py b/CodeXL/Components/Graphics/GPS_init.py
index 8b42027..805191d 100644
--- a/CodeXL/Components/Graphics/GPS_init.py
+++ b/CodeXL/Components/Graphics/GPS_init.py
@@ -47,16 +47,16 @@ def ParseCommandLine(env, buildInternal):
 # initialize Gtk library
 def initGtk(env):
     env.Append(BASE_PATH = [
-        "/usr/include/cairo",
-        "/usr/include/pango-1.0",
-        "/usr/include/atk-1.0",
-        "/usr/include/gtk-2.0",
-        "/usr/include/gdk-pixbuf-2.0",
-        "/usr/include/glib-2.0",
-        "/usr/lib64/gtk-2.0/include",
-        "/usr/lib64/glib-2.0/include/",
-        "/usr/lib/x86_64-linux-gnu/glib-2.0/include",
-        "/usr/lib/x86_64-linux-gnu/gtk-2.0/include",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/cairo",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/pango-1.0",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/atk-1.0",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/gtk-2.0",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/gdk-pixbuf-2.0",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/glib-2.0",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/lib64/gtk-2.0/include",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/lib64/glib-2.0/include/",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/lib/x86_64-linux-gnu/glib-2.0/include",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/lib/x86_64-linux-gnu/gtk-2.0/include",
    ])
 
 ##########################################################################################

diff --git a/Common/Src/SCons/CXL_init.py b/Common/Src/SCons/CXL_init.py
index bf36509..0672013 100755
--- a/Common/Src/SCons/CXL_init.py
+++ b/Common/Src/SCons/CXL_init.py
@@ -433,17 +433,17 @@ def UseQtNoGraphics(env):
 
 def initGtk (env) :
     env.Append(CXL_Gtk_inc_path = [
-        "/usr/include/cairo",
-        "/usr/include/pango-1.0",
-        "/usr/include/atk-1.0",
-        "/usr/include/gtk-2.0",
-        "/usr/include/gdk-pixbuf-2.0",
-        "/usr/include/glib-2.0",
-        "/usr/lib64/gtk-2.0/include",
-        "/usr/lib64/glib-2.0/include/",
-    "/usr/lib/x86_64-linux-gnu/glib-2.0/include",
-    "/usr/lib/x86_64-linux-gnu/gtk-2.0/include",
-    "/usr/include/x86_64-linux-gnu/",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/cairo",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/pango-1.0",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/atk-1.0",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/gtk-2.0",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/gdk-pixbuf-2.0",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/glib-2.0",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/lib64/gtk-2.0/include",
+        env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/lib64/glib-2.0/include/",
+    env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/lib/x86_64-linux-gnu/glib-2.0/include",
+    env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/lib/x86_64-linux-gnu/gtk-2.0/include",
+    env['PKG_CONFIG_SYSROOT_DIR'] + "/usr/include/x86_64-linux-gnu/",
     ])
     # No additional libraries needed for Gtk
 
-- 
1.9.1

